src/dataProvider/SparqlDataProvider.js

```diff
@@ -32,22 +79,42 @@ export default {
     query.offset = (pagination.page - 1) * pagination.perPage;
     query.sort = sort;
 
+    handleComunicaContextCreation(query);
+
+    if (query.sourcesIndex) {
+      try {
+        const additionalSources = await addComunicaContextSourcesFromSourcesIndex(query.sourcesIndex);
+        query.comunicaContext.sources = [...new Set([...query.comunicaContext.sources, ...additionalSources])];
+      } catch (error) {
+        console.error("Error fetching sources from index:", error);
+        throw error;
+      }
+    }
+
     if (meta && meta.variables) {
-      query.variableValues = meta.variables
+      query.variableValues = meta.variables;
+    }
+
+    let results;
+    try {
+      results = await executeQuery(query);
+    } catch (error) {
+      console.error("Error executing query:", error);
```

Comment:
This code change is not useful. I guess it's a remainder of a debugging session. Avoid writing to the console in finished code.